esp8266/Makefile

```diff
@@ -148,7 +148,7 @@ DRIVERS_SRC_C = $(addprefix drivers/,\
 SRC_S = \
 	gchelper.s \
 
-FROZEN_MPY_PY_FILES := $(shell find -L $(FROZEN_MPY_DIR) -type f -name '*.py')
+FROZEN_MPY_PY_FILES := $(shell find -L $(FROZEN_MPY_DIR) -type f -name '*.py') modules/upip.py modules/upip_utarfile.py
```

Comment:
> Perhaps I'm asking too complex questions

No it's ok, I just missed the point where the symlink would be in the git repo and thought it would be created in the makefile. Looking around for an answer it is basically e): should work but depends on certain conditions: https://github.com/git-for-windows/git/wiki/Symbolic-Links. 

>  just a problem of the Windows maintainer

..or the windows user in this case. Anyway it's ok for me. I cloned a test repo using ```git clone -c core.symlinks=true  https://github.com/ariofrio/symlink-test``` and that works fine and creates file symlinks. So if it ever causes problems for someone else the answer is gonna be 'micropython side is fine, your machine isn't'.